### PR TITLE
M2 3 5 image annotation

### PR DESCRIPTION
**Pull request by FabianMauz**
- [ ] rather 1-story 1-commit than sub-atomic commits

- [ ] commit title is meaningful =>  git history search

- [ ] commit description is helpful => helps the reviewer to understand the changes

- [ ] code is up-to-date with the latest developments of the target branch (rebased to it or whatever) => :fast_forward:-merge for linear history is favoured

- [ ] added code is linted

- [ ] tests are passing (at least locally): we still have some random test failure on CI. thinking of asking spec/examples.txt to be commited

- [ ] in case the changes are visible to the end-user,  video or screenshots should be added to the PR => helps  with user testing

- [ ] testing coverage improvement is improved.

- [ ] CHANGELOG :  add a bullet point on top (optional: reference to github issue/PR )

- [ ] parallele PR for documentation  on docusaurus  if the feature/fix is tagged for a release

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -178,6 +178,11 @@ vendor/cache/*
 shrine/*
 .docker-sync/
 
+shrine/*
```
Comment (JanCBrammer) [inline .gitignore:181]: Duplicate entry, also specified on line 178.

File: app/api/chemotion/attachment_api.rb [hunk 2]
```diff
@@ -254,18 +297,15 @@ def upload_chunk_error_message
       params do
         requires :attachment_id, type: Integer, desc: 'Database id of image attachment'
         optional :identifier, type: String, desc: 'Identifier(UUID) of image attachment as fallback loading criteria'
+        optional :annotated, type: Boolean, desc: 'Return annotated image if possible'
       end
 
       get 'image/:attachment_id' do
```
Comment (JanCBrammer) [inline app/api/chemotion/attachment_api.rb:303]: `desc` and `params` missing.

File: app/packs/src/apps/mydb/elements/details/researchPlans/SaveResearchPlanWarning.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+SaveResearchPlanWarning.propTypes = {
+    visible: PropTypes.bool.isRequired,
+    warningStyle: PropTypes.string,
+    warningMessage: PropTypes.string
+};
+
+SaveResearchPlanWarning.defaultProps={
+    warningStyle : 'imageEditedWarning',
+    warningMessage: 'Image was edited. Please save Researchplan to apply changes'
```
Comment (JanCBrammer) [inline app/packs/src/apps/mydb/elements/details/researchPlans/SaveResearchPlanWarning.js:31]: Missing period at end of warning.

File: app/usecases/attachments/annotation/annotation_updater.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+          sanitized_svg_string = sanitize_svg_string(annotation_svg_string)
+          save_svg_string_to_file_system(sanitized_svg_string, attachment)
+          update_thumbnail(attachment, sanitized_svg_string)
+          create_annotated_flat_image(attachment, sanitized_svg_string)
+        end
+
+        def sanitize_svg_string(svg_string) # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
+          scrubber = Rails::Html::PermitScrubber.new
+          scrubber.tags = %w[svg image g title rect text path line ellipse]
+          scrubber.attributes = %w[height id width href class fill stroke stroke-dasharray stroke-linecap
```
Comment (JanCBrammer) [inline app/usecases/attachments/annotation/annotation_updater.rb:26]: Allow rotation.

File: db/structure.sql [hunk 5]
```diff
@@ -0,0 +1,6361 @@
+SET statement_timeout = 0;
```
Comment (JanCBrammer) [inline db/structure.sql:1]: Remove this file.

**Review by JanCBrammer — COMMENTED** [review]
WIP

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldImage.js [hunk 6]
```diff
@@ -109,6 +148,36 @@ export default class ResearchPlanDetailsFieldImage extends Component {
...
   }
 
+  renderImageEditedWarning(attachment){
+    if(!attachment) {return null;}
+    if(!attachment.updatedAnnotation){return null;} 
+    
+    return (
+      <div className='imageEditedWarning'>
+        <Alert>Image was edited. Please save Researchplan to apply changes</Alert>
+      </div>
```
Comment (JanCBrammer) [inline app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldImage.js:158]: Use `app/packs/src/apps/mydb/elements/details/researchPlans/SaveResearchPlanWarning.js`?

**Review by JanCBrammer — COMMENTED** [review]
What's the purpose of `db/structure.sql` (equivalent of `db/schema.rb`) and `shipping_label.gif`? Should those files be removed?

File: app/usecases/attachments/annotation/annotation_loader.rb [hunk 7]
```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+# Class for loading an annotation of an attachment with the shrine infrastructure
+module Usecases
+  module Attachments
+    module Annotation
+      class AnnotationLoader
+        def get_annotation_of_attachment(attachment_id) # rubocop:disable Metrics/AbcSize
+          att = Attachment.find(attachment_id)
+          raise 'could not find annotation' unless att
```
[the remark this answers is not in the document]
Comment (FabianMauz) [reply app/usecases/attachments/annotation/annotation_loader.rb:10]: You are right, i changed it

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldImage.js (hunk 6 above)
Comment (FabianMauz) [reply app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldImage.js:158]: The hole function was replaced by the SaveResearchPlanWarning component and is now unused, I removed it.

File: app/api/chemotion/attachment_api.rb (hunk 2 above)
Comment (FabianMauz) [reply app/api/chemotion/attachment_api.rb:303]: added

**Pull request merged**